### ISSUE TITLE
Bug: `contains: "%"` always matches every row, but it shouldn't

### ISSUE THREAD

**Issue by dmarcelino** [balderdashy/waterline-sequel#33]
*SQL adapters `contains`' criteria acts as a `LIKE '%[value]%'` but it shouldn't as the documentation does not imply `contains` is meant to take SQL syntax:

From the documentation:
> #### 'contains'
> A shorthand for pattern matching both sides of a string. Will return records where the value contains the string anywhere inside of it.

Given `contains` is meant to be consistent across adapters, the value should be escaped in SQL adapters. For example `Model.find().where({name: {contains: "%"}}` should become `select * where name like '%\%%'`.

The issue seems to lie in the criteria processor here:
https://github.com/balderdashy/waterline-sequel/blob/master/sequel/lib/criteriaProcessor.js#L623-L645

Originally raised by @cspotcode  in balderdashy/waterline#899 (click for details and discussion).

**Comment by sailsbot** [comment]
Thanks for posting, @dmarcelino.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by AmirTugi** [comment]
@dmarcelino  Has there been any progress regarding this issue? This is still an issue on waterline-sequel 0.6.4

**Comment by dmarcelino** [comment]
Hey @AmirTugi, you'll have to ask someone from the sails team.

**Comment by AmirTugi** [comment]
So I sat around trying to to see where the issue starts from, and I realized the problem was actually not a problem in the code, but rather I did not encode the URL I used to retrieve the data by SailsJS blueprints routes. So after I encoded it, everything worked like a charm.
Thanks anyway
